compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -77,6 +83,42 @@ generateSrcFromLf env = noLoc mod
         let occName = mkOccName varName $ T.unpack $ sanitize dataTypeCon0
         pure $ mkDataDecl env thisModule occName dataParams (convDataCons dataTypeCon0 dataCons)
 
+    valueDecls = do
+        dval@LF.DefValue {..} <- NM.toList $ LF.moduleValues $ envMod env
+        guard $ shouldExposeDefValue dval
+        let (lfName, lfType) = dvalBinder
+            ltype = noLoc $ convType env lfType :: LHsType GhcPs
+            lname = mkRdrName (LF.unExprValName lfName) :: Located RdrName
+            sig = TypeSig noExt [lname] (HsWC noExt $ HsIB noExt ltype)
+            lsigD = noLoc $ SigD noExt sig :: LHsDecl GhcPs
+            lexpr = noLoc $ HsPar noExt $ noLoc $ HsVar noExt lname :: LHsExpr GhcPs
+            lgrhs = noLoc $ GRHS noExt [] lexpr :: LGRHS GhcPs (LHsExpr GhcPs)
+            grhss = GRHSs noExt [lgrhs] (noLoc $ EmptyLocalBinds noExt)
+            matchContext = FunRhs lname Prefix NoSrcStrict
+            lmatch = noLoc $ Match noExt matchContext [] Nothing grhss
+            lalts = noLoc [lmatch]
+            bind = FunBind noExt lname (MG noExt lalts Generated) WpHole []
+            lvalD = noLoc $ ValD noExt bind :: LHsDecl GhcPs
+        [ lsigD, lvalD ]
+
+    shouldExposeDefDataType :: LF.DefDataType -> Bool
+    shouldExposeDefDataType LF.DefDataType{..}
+        = LF.getIsSerializable dataSerializable
+        || not (LF.moduleNameString lfModName == "GHC.Prim")
+
+    shouldExposeDefValue :: LF.DefValue -> Bool
+    shouldExposeDefValue LF.DefValue{..}
+        | (lfName, lfType) <- dvalBinder
+        , lfNameText <- LF.unExprValName lfName
+        = not (LF.getIsTest dvalIsTest)
+        && not (T.null lfNameText)
```

Comment:
Do we actually generate definitions with an empty name? I thought that wasn’t even allowed in DAML-LF.

Reply:
I removed this check anyway :) It's back to being

```Haskell
not ("$" `T.isPrefixOf` LF.unExprValName lfName)
```